src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -416,21 +418,16 @@ struct
       set t (Location.Account addr) account
 
     let to_list t =
-      let mask_accounts = find_all_accounts t in
-      let parent_accounts = Base.to_list (get_parent t) in
-      let mask_keys =
-        List.map mask_accounts ~f:(fun acct -> Account.public_key acct)
-      in
-      let mask_key_set = Key.Set.of_list mask_keys in
-      (* if an account is in mask and parent, favor the mask version *)
-      let not_in_mask parent_account =
-        let parent_key = Account.public_key parent_account in
-        not (Key.Set.mem mask_key_set parent_key)
-      in
-      let in_parent_not_in_mask_accounts =
-        List.filter parent_accounts ~f:not_in_mask
-      in
-      mask_accounts @ in_parent_not_in_mask_accounts
+      keys t |> Set.to_list
+      |> List.map ~f:(fun key ->
+             let location = location_of_key t key |> Option.value_exn in
+             match location with
+             | Account addr ->
+                 (Addr.to_int addr, get t location |> Option.value_exn)
+             | location -> raise (Location_is_not_account location) )
```

Comment:
should this be a `failwith`, since it indicates an incoherence in the mask tables?

Reply:
I was thinking about that, but I see `failwith` in our codebase associated with code that is not implemented. I see exceptions/or_error involving invariances being violated in our code.

Reply:
I see it being used both ways.